src/test/unit/specific_feature_tests/dialogs.js

```diff
@@ -0,0 +1,423 @@
+            );
+        }
+        ok(
+            dialogWindow.window === dialogWindow,
+            "Dialog window seems to exist"
+        );
+
+        strictEqual(
+            dialogWindow.document.title,
+            args.expectedTitle,
+            "Dialog title"
+        );
+    }
+    function assertNotOpened(returnedFalse) {
+        expect(expect() + 1);
+        if (sinonCantWrapWindowOpen !== true) {
+            expect(expect() + 1);
+            ok(
+                window.open.callCount === 0,
+                "`window.open` was not called"
+            );
+        }
+        ok(
+            returnedFalse === false,
+            "Dialog function returned `false`"
+        );
+    }
+}
+/* jshint latedef: true */
+
+dialogTestHelper.EXPECT_OPENED = "Expect this dialog to have been opened. " +
+    "Expect it with all of your heart.";
+dialogTestHelper.EXPECT_NOT = "Nope. The dialog should not have opened.";
+
+function getDialogToolbarSelector(dialogName) {
+    if (typeof dialogName !== "string") {
+        throw "Expected a string";
+    }
+    return ".wym_section.wym_tools.wym_buttons li a[name=" + dialogName + "]";
+}
+
+function getDialogDocumentTitle(command) {
+    if (typeof command !== "string") {
+        throw "Expected a string";
+    }
+
+    var TITLES = {
+        CreateLink: "Link",
+        InsertImage: "Image",
+        Paste: "Paste from Word",
+        InsertTable: "Table",
+        Preview: "Preview"
+    };
+
+    if (TITLES.hasOwnProperty(command) !== true) {
+        throw "No such dialog";
+    }
+    return TITLES[command];
+}
+
+test("Link dialog doesn't open when no selection; by toolbar button click",
+    function () {
+    var command = WYMeditor.EXEC_COMMANDS.CREATE_LINK;
+    dialogTestHelper({
+        noChangeHtml: "<p>Foo</p>",
+        prepareFunc: function (wymeditor) {
+            wymeditor.deselect();
+        },
+        dialogButtonSelector: getDialogToolbarSelector(command),
+        expectedOpenedOrNot: dialogTestHelper.EXPECT_NOT,
+        expectedTitle: getDialogDocumentTitle(command)
+    });
+});
+
+test("Link dialog opens when selection is non-collapsed and " +
+    "`selectedContainer` doesn't return false; by toolbar button click",
+    function () {
+    var command = WYMeditor.EXEC_COMMANDS.CREATE_LINK;
+    dialogTestHelper({
+        noChangeHtml: "<p>Foo</p>",
+        prepareFunc: function (wymeditor) {
+            var p = wymeditor.body().childNodes[0];
+            makeTextSelection(
+                wymeditor,
+                p,
+                p,
+                0,
+                3
+            );
+        },
+        dialogButtonSelector: getDialogToolbarSelector(command),
+        expectedOpenedOrNot: dialogTestHelper.EXPECT_OPENED,
+        expectedTitle: getDialogDocumentTitle(command)
+    });
+});
+
+test("Link dialog doesn't open when selection is collapsed and " +
+    "`selectedContainer` doesn't return false; by toolbar button click",
```

Comment:
I think it's important to point out that the `selectedContainer` isn't an anchor, here. So maybe:
and selectedContainer is a non-anchor element
instead of 
and selectedContainer doesn't return false